GoogleSignIn/Tests/Unit/GIDAuthorizationResponseHelperTest.m

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import <XCTest/XCTest.h>
+
+#import "GoogleSignIn/Sources/GIDAuthorizationResponse/GIDAuthorizationResponseHelper.h"
+#import "GoogleSignIn/Sources/GIDAuthorizationResponse/Fake/GIDAuthorizationResponseHandlingFake.h"
+#import "GoogleSignIn/Sources/GIDAuthorizationResponse/Implementations/GIDAuthorizationResponseHandler.h"
+
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDConfiguration.h"
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDSignIn.h"
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h"
+#import "GoogleSignIn/Sources/GIDAuthFlow.h"
+
+#import "GoogleSignIn/Tests/Unit/OIDAuthorizationResponse+Testing.h"
+#import "GoogleSignIn/Tests/Unit/OIDAuthState+Testing.h"
+
+// The EMM support version
+static NSString *const kEMMVersion = @"1";
+
+@interface GIDAuthorizationResponseHelperTest : XCTestCase
+@end
+
+@implementation GIDAuthorizationResponseHelperTest
+
+- (void)testInitWithAuthorizationResponseHandler {
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:nil
+                                                                    flowName:GIDFlowNameVerify
+                                                               configuration:nil
+                                                                       error:nil];
+  GIDAuthorizationResponseHelper *responseHelper = 
+      [[GIDAuthorizationResponseHelper alloc] initWithAuthorizationResponseHandler:responseHandler];
+
+  XCTAssertNotNil(responseHelper);
+  XCTAssertNotNil(responseHelper.responseHandler);
+}
+
+- (void)testFetchTokenWithAuthFlow {
+  OIDAuthState *authState = [OIDAuthState testInstance];
+  GIDAuthorizationResponseHandlingFake *fakeHandler = 
+      [[GIDAuthorizationResponseHandlingFake alloc] initWithAuthState:authState error:nil];
+  GIDAuthorizationResponseHelper *responseHelper =
+      [[GIDAuthorizationResponseHelper alloc] initWithAuthorizationResponseHandler:fakeHandler];
+  GIDAuthFlow *authFlow = [[GIDAuthFlow alloc] init];
+
+  [responseHelper fetchTokenWithAuthFlow:authFlow];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNotNil(authFlow.authState);
```

Comment:
Assert equality. Not nil is useful, but it's more important to know that the `authState` on line 53 is the same as `authState.authState` here.

Reply:
Gotcha, done.